### PR TITLE
simplify docker quickstart

### PR DESCRIPTION
**Pull request by gpene**
The idea here is the easiest possible version of getting nakama running in Docker - I've seen several mentions in our forum/slack of >50% of gamedevs saying they have no docker experience and struggling to get started.

Docker-compose, additional configs, and more advanced options (like binding mounts) can be covered in a separate section at a later time. For this the goal is simply that immediate sense of success/accomplishment in getting it installed and started.

### CODE REVIEW

File: docs/install-docker-quickstart.md [hunk 1]
```diff
@@ -1,264 +1,84 @@
-# Docker quickstart
+# Docker Quickstart
 
-To start developing with Nakama, you’ll first need to install it on your development machine. It’s straightforward and takes just a few minutes. In this guide we’re going to focus on installing your Nakama development instance using Docker.
+[Docker](https://www.docker.com/) is the quickest way to download and start developing with Nakama. By using Docker you are able to:
 
-!!! summary "Recommended Approach"
-    Docker is the quickest way to download and get started with Nakama for development purposes. For production settings, we recommend that you install Nakama as a [binary](install-binary.md) to ensure all system resources are available to Nakama.
+* Install to a pristine environment
+* Concurrently install and run the required [CockroachDB](https://www.cockroachlabs.com/) server
```
Comment (mofirouz) [inline docs/install-docker-quickstart.md:6]: >[CockroachDB](https://www.cockroachlabs.com/) server
->
>[CockroachDB](https://www.cockroachlabs.com/) database

**Review by mofirouz — CHANGES_REQUESTED** [review]
Generally like the simplification (as most of the docker instruction have been mostly outdated by now). 

However, we find that asking the developer to follow the docker-compose installation has been much more successful and easier for them to do. I don't know if you were planning on having another page for it but if I were to prioritise installation guides I'd say:

1. Docker-compose (and special instructions needed for Windows) - incl how to feed Nakama a custom config file.
2. Docker
3. Binary installation

Finally the latest Docker-compose page is on GitHub:
https://github.com/heroiclabs/nakama/blob/master/docker-compose.yml
https://github.com/heroiclabs/nakama/blob/master/docker-compose-postgres.yml

If we could somehow source the content from GitHub file and display in-line in the docs that would be a massive win!

Overall, good work, thank you.

**Comment by gpene** [comment]
@mofirouz Got it - I'll rework this PR to prioritize the docker-compose install 👍 Thanks!

File: docs/install-docker-quickstart.md [hunk 2]
```diff
@@ -1,264 +1,84 @@
-# Docker quickstart
+# Docker Quickstart
 
-To start developing with Nakama, you’ll first need to install it on your development machine. It’s straightforward and takes just a few minutes. In this guide we’re going to focus on installing your Nakama development instance using Docker.
+[Docker](https://www.docker.com/) is the quickest way to download and start developing with Nakama. By using Docker you are able to:
 
-!!! summary "Recommended Approach"
-    Docker is the quickest way to download and get started with Nakama for development purposes. For production settings, we recommend that you install Nakama as a [binary](install-binary.md) to ensure all system resources are available to Nakama.
+* Install to a pristine environment
+* Concurrently install and run the required [CockroachDB](https://www.cockroachlabs.com/) database
```
Comment (seanpackham) [inline docs/install-docker-quickstart.md:6]: I think we can drop "required" as Postgres is also supported.
Minor, what do you think of changing "concurrently" to "easily" as it speaks to how Docker simplifies complex tech which stands out more than doing something concurrently?

File: docs/install-docker-quickstart.md [hunk 3]
```diff
@@ -1,264 +1,84 @@
...
 
-=== "Shell"
-	```shell
-	docker run heroiclabs/nakama <command> [options]
-	```
+## Prerequisites
 
-!!! note "Database server"
-    Nakama requires a database server running. Make sure you start the database before Nakama, or [use docker-compose to run both](#running-nakama-with-docker-compose).
+Before proceeding ensure that you have [installed Docker Desktop](https://docs.docker.com/get-docker/).
```
Comment (seanpackham) [inline docs/install-docker-quickstart.md:14]: Could you add a note that you need x, y, z operating system. For example Win 10 Home is not supported and going down that path might frustrate someone new to Nakama.

File: docs/install-docker-quickstart.md [hunk 4]
```diff
@@ -1,264 +1,84 @@
...
 
-**Host**: `127.0.0.1` (or `localhost`)
-**Port** : `7350`
-**SSL**: `False`
-**Server Key**: `defaultkey`
+![Nakama console](images/console.png)
 
-In the JavaScript client, you can create a `client` like this:
+!!! note "Note"
+    When prompted to login, the default credentials are `Admin:Password`. These can be changed via configuration file or command-line flags.
```
Comment (seanpackham) [inline docs/install-docker-quickstart.md:72]: Thanks for adding this 😄
Are the first letters capatalised?

File: docs/install-docker-compose-quickstart.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+6. Congratulations! Your Nakama server is now up and running, available at `127.0.0.1:7350`.
+
+![Nakama containers running](images/nakama-compose-running.png)
+
+Use the **Open in Visual Studio Code** button (or that for your IDE) to edit your `docker-compose.yml` file directly.
+
+## Nakama Console
+
+You can also access the Nakama Console by navigating your browser to `localhost:7351`:
```
Comment (mofirouz) [inline docs/install-docker-compose-quickstart.md:46]: In some OSs, `localhost` acts out and doesn't resolve - it's always better to refer with `127.0.0.1`

File: docs/install-docker-compose-quickstart.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+## Prerequisites
+
+Before proceeding ensure that you have [installed Docker Desktop](https://docs.docker.com/get-docker/).
+
+## Running Nakama
+
+1. Start by creating a directory where your Nakama server will sit, for example `Desktop/nakama`.
+2. In this folder create a `docker-compose.yml` file and open it using your preferred text editor.
+3. Heroic Labs provides two YML files for use: using either [CockroachDB](https://github.com/heroiclabs/nakama/blob/master/docker-compose.yml) or [PostgreSQL](https://github.com/heroiclabs/nakama/blob/master/docker-compose-postgres.yml) as the database.
```
Comment (mofirouz) [inline docs/install-docker-compose-quickstart.md:20]: 1. Can we use embed these resources on the page directly from GitHub, some type of embedding by any chance?
2. We should probably edit these to include a `volume` and load a custom config file (with a really simple config) which will significantly make following "## Configuration File" much easier to follow IMO.

File: docs/install-docker-compose-quickstart.md [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    When prompted to login, the default credentials are `admin:password`. These can be changed via configuration file or command-line flags.
+
+## Configuration File
+
+There are many [configuration options](install-configuration.md) available that you can customize for your Nakama server. You can create a YML file for all configurations you want to set, and pass that file to your Docker containers.
+
+First you will need to make a local storage volume available to Docker:
+
+1. Open your `docker-compose.yml` file in your preferred text editor.
+2. Edit the `nakama:volumes:` entry to specify your desired volume. For example, to create a `/data` folder in our `desktop/nakama` directory used above, which would be available at `nakama/data` in the Docker container, it would look like the following:
```
Comment (mofirouz) [inline docs/install-docker-compose-quickstart.md:60]: Left a note above about making this easier to follow.

File: docs/install-docker-quickstart.md (hunk 3 above)
Comment (mofirouz) [reply docs/install-docker-quickstart.md:14]: Yep agreed.

File: docs/install-docker-quickstart.md [hunk 8]
```diff
@@ -1,264 +1,83 @@
...
-	      - "7351:7351"
-	    healthcheck:
-	      test: ["CMD", "curl", "-f", "http://localhost:7350/"]
-	      interval: 10s
-	      timeout: 5s
-	      retries: 5
-	volumes:
-	  data:
-	```
+![Pull docker images](images/docker-pull-result.png)
```
Comment (mofirouz) [inline docs/install-docker-quickstart.md:29]: nit: about image naming - do you think it's better to put the images used in this document in its own folder (that is named the same as the file name)?

**Review by mofirouz — COMMENTED** [review]
Nice one, some small changes to do with config file loading.

File: docs/install-docker-quickstart.md (hunk 3 above)
Comment (gpene) [reply docs/install-docker-quickstart.md:14]: Added a note about Linux installs. Why do you say that Win 10 Home isn't supported: 
Screen Shot 2021-05-12 at 10 29 47 AM

File: docs/install-docker-compose-quickstart.md (hunk 6 above)
Comment (gpene) [reply docs/install-docker-compose-quickstart.md:23]: One option that showed up after some searching: https://www.npmjs.com/package/embedme
maybe @seanpackham knows some other options here?

File: docs/install-docker-quickstart.md (hunk 8 above)
Comment (gpene) [reply docs/install-docker-quickstart.md:29]: Agreed - pretty trivial now but definitely better as the amount of content grows. Done 👍

File: docs/install-docker-quickstart.md (hunk 3 above)
Comment (seanpackham) [reply docs/install-docker-quickstart.md:14]: Windows 10 Home doesn't have Hyper-V which is needed for Docker or WSL

File: docs/install-docker-compose-quickstart.md (hunk 6 above)
Comment (seanpackham) [reply docs/install-docker-compose-quickstart.md:23]: Shared a mkdocs extension with Gabe https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#snippets

File: docs/install-docker-quickstart.md (hunk 3 above)
Comment (gpene) [reply docs/install-docker-quickstart.md:14]: No longer the case with WSL 2: https://www.docker.com/blog/docker-desktop-for-windows-home-is-here/

Comment (seanpackham) [reply docs/install-docker-quickstart.md:14]: Is that out of Windows Insider preview? I tried it a couple weeks ago and wasn't successful.

Comment (seanpackham) [reply docs/install-docker-quickstart.md:14]: After some debugging and changing a BIOS setting, I got wsl-2 and Docker Desktop running on WIndows 10 Home. Thanks @gpene 🙏

File: docs/install-docker-compose-quickstart.md (hunk 6 above)
Comment (gpene) [reply docs/install-docker-compose-quickstart.md:23]: @mofirouz There is no immediately available solution for this right now (that wouldn't involve some level of duplication, defeating the purpose), but did discuss with Sean and it is something we will be able to address in v2 of the doc site.

I think let's merge this as-is now, and improve upon it in v2.

**Comment by seanpackham** [comment]
@gpene running locally I got these info/warning messages:
-  The following pages exist in the docs directory, but are not included in the "nav" configuration: install-docker-quickstart.md
- Documentation file 'install-docker-quickstart.md' contains a link to 'images\console.png' which is not found in the documentation files.

**Comment by gpene** [comment]
> @gpene running locally I got these info/warning messages:
> 
> * The following pages exist in the docs directory, but are not included in the "nav" configuration: install-docker-quickstart.md
> * Documentation file 'install-docker-quickstart.md' contains a link to 'images\console.png' which is not found in the documentation files.

Yep, I should've been clearer when I mentioned above, that's the page for the non-compose Docker install - I left it out of the nav configuration because I don't think we should/need to provide two differing examples when the idea is simplification. But I saved the file incase you guys disagree and we can expand and add it back later. The broken image link is for this same page so it's not an issue - also the image is there, just in another directory.

**Comment by seanpackham** [comment]
@gpene good article, easy to follow and I like the inclusion of postgres and supplying a configuration file to Docker.

**Comment by seanpackham** [comment]
@mofirouz ready for you to review.

**Review by mofirouz — APPROVED** [review]

**Review by seanpackham — APPROVED** [review]

**Comment by seanpackham** [comment]
Imported to Docs Cetre repo, closing this issue.

**Pull request closed without merging**